openedx/core/djangoapps/content_libraries/api/libraries.py

```diff
@@ -692,3 +689,42 @@ def revert_changes(library_key: LibraryLocatorV2, user_id: int | None = None) ->
 
     # Call the event handlers as needed.
     tasks.wait_for_post_revert_events(draft_change_log, library_key)
+
+
+def get_backup_task_status(
+    library_key: LibraryLocatorV2,
+    user_id: int,
+    task_id: str | None = None
+) -> dict | None:
+    """
+    Get the status of a library backup task.
+
+    Returns a dictionary with the following keys:
+        - state: One of "Pending", "Exporting", "Succeeded", "Failed"
+        - url: If state is "Succeeded", the URL where the exported .zip file can be downloaded. Otherwise, None.
+    If no task is found, returns None.
+    """
+    task_status = None
+    if not task_id:
```

Comment:
[Nit]: In general, if you have the choice between "if condition, else:" or "if not condition, else", do the former, i.e. make this clause `if task_id:`